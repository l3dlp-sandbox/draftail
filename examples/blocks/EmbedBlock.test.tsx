import React from "react";
import { shallow } from "enzyme";

import { ContentBlock, EditorState } from "draft-js";
import EmbedBlock from "./EmbedBlock";

describe("EmbedBlock", () => {
  it("renders", () => {
    expect(
      shallow(
        <EmbedBlock
          block={new ContentBlock()}
          blockProps={{
            entityType: {
              description: "test",
              icon: "test",
            },
            editorState: EditorState.createEmpty(),
            entityKey: "string",
            textDirectionality: "RTL",
            lockEditor: () => {},
            unlockEditor: () => {},
            onChange: () => {},
            onEditEntity: () => {},
            onRemoveEntity: () => {},
            entity: {
              getType: () => "type",
              getMutability: () => "MUTABLE",
              getData: () => ({
                url: "http://www.example.com/",
                title: "Test title",
                thumbnail: "http://www.example.com/example.png",
              }),
            },
          }}
        />,
      ),
    ).toMatchSnapshot();
  });

  it("no data", () => {
    expect(
      shallow(
        <EmbedBlock
          block={new ContentBlock()}
          blockProps={{
            entityType: {
              description: "test",
              icon: "test",
            },
            editorState: EditorState.createEmpty(),
            entityKey: "string",
            textDirectionality: "RTL",
            lockEditor: () => {},
            unlockEditor: () => {},
            onChange: () => {},
            onEditEntity: () => {},
            onRemoveEntity: () => {},
            entity: {
              getType: () => "type",
              getMutability: () => "MUTABLE",
              getData: () => ({}),
            },
          }}
        />,
      ),
    ).toMatchSnapshot();
  });
});
