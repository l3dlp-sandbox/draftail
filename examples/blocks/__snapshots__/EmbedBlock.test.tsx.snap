// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmbedBlock no data 1`] = `
<MediaBlock
  block={
    Immutable.Record {
      "key": "",
      "type": "unstyled",
      "text": "",
      "characterList": Immutable.List [],
      "depth": 0,
      "data": Immutable.Map {},
    }
  }
  blockProps={
    Object {
      "editorState": EditorState {
        "_immutable": Immutable.Record {
          "allowUndo": true,
          "currentContent": Immutable.Record {
            "entityMap": Object {
              "__add": [Function],
              "__create": [Function],
              "__get": [Function],
              "__getLastCreatedEntityKey": [Function],
              "__mergeData": [Function],
              "__replaceData": [Function],
              "add": [Function],
              "create": [Function],
              "get": [Function],
              "getLastCreatedEntityKey": [Function],
              "mergeData": [Function],
              "replaceData": [Function],
            },
            "blockMap": Immutable.OrderedMap {
              "5h4sv": Immutable.Record {
                "key": "5h4sv",
                "type": "unstyled",
                "text": "",
                "characterList": Immutable.List [],
                "depth": 0,
                "data": Immutable.Map {},
              },
            },
            "selectionBefore": Immutable.Record {
              "anchorKey": "5h4sv",
              "anchorOffset": 0,
              "focusKey": "5h4sv",
              "focusOffset": 0,
              "isBackward": false,
              "hasFocus": false,
            },
            "selectionAfter": Immutable.Record {
              "anchorKey": "5h4sv",
              "anchorOffset": 0,
              "focusKey": "5h4sv",
              "focusOffset": 0,
              "isBackward": false,
              "hasFocus": false,
            },
          },
          "decorator": null,
          "directionMap": Immutable.OrderedMap {
            "5h4sv": "LTR",
          },
          "forceSelection": false,
          "inCompositionMode": false,
          "inlineStyleOverride": null,
          "lastChangeType": null,
          "nativelyRenderedContent": null,
          "redoStack": Immutable.Stack [],
          "selection": Immutable.Record {
            "anchorKey": "5h4sv",
            "anchorOffset": 0,
            "focusKey": "5h4sv",
            "focusOffset": 0,
            "isBackward": false,
            "hasFocus": false,
          },
          "treeMap": Immutable.OrderedMap {
            "5h4sv": Immutable.List [
              Immutable.Record {
                "start": 0,
                "end": 0,
                "decoratorKey": null,
                "leaves": Immutable.List [
                  Immutable.Record {
                    "start": 0,
                    "end": 0,
                  },
                ],
              },
            ],
          },
          "undoStack": Immutable.Stack [],
        },
      },
      "entity": Object {
        "getData": [Function],
        "getMutability": [Function],
        "getType": [Function],
      },
      "entityKey": "string",
      "entityType": Object {
        "description": "test",
        "icon": "test",
      },
      "lockEditor": [Function],
      "onChange": [Function],
      "onEditEntity": [Function],
      "onRemoveEntity": [Function],
      "textDirectionality": "RTL",
      "unlockEditor": [Function],
    }
  }
  isLoading={true}
>
  <a
    className="EmbedBlock__link"
    rel="noopener noreferrer"
    target="_blank"
  />
  <button
    className="Tooltip__button"
    onClick={[Function]}
    type="button"
  >
    Edit
  </button>
  <button
    className="Tooltip__button"
    onClick={[Function]}
    type="button"
  >
    Remove
  </button>
</MediaBlock>
`;

exports[`EmbedBlock renders 1`] = `
<MediaBlock
  block={
    Immutable.Record {
      "key": "",
      "type": "unstyled",
      "text": "",
      "characterList": Immutable.List [],
      "depth": 0,
      "data": Immutable.Map {},
    }
  }
  blockProps={
    Object {
      "editorState": EditorState {
        "_immutable": Immutable.Record {
          "allowUndo": true,
          "currentContent": Immutable.Record {
            "entityMap": Object {
              "__add": [Function],
              "__create": [Function],
              "__get": [Function],
              "__getLastCreatedEntityKey": [Function],
              "__mergeData": [Function],
              "__replaceData": [Function],
              "add": [Function],
              "create": [Function],
              "get": [Function],
              "getLastCreatedEntityKey": [Function],
              "mergeData": [Function],
              "replaceData": [Function],
            },
            "blockMap": Immutable.OrderedMap {
              "cvvlm": Immutable.Record {
                "key": "cvvlm",
                "type": "unstyled",
                "text": "",
                "characterList": Immutable.List [],
                "depth": 0,
                "data": Immutable.Map {},
              },
            },
            "selectionBefore": Immutable.Record {
              "anchorKey": "cvvlm",
              "anchorOffset": 0,
              "focusKey": "cvvlm",
              "focusOffset": 0,
              "isBackward": false,
              "hasFocus": false,
            },
            "selectionAfter": Immutable.Record {
              "anchorKey": "cvvlm",
              "anchorOffset": 0,
              "focusKey": "cvvlm",
              "focusOffset": 0,
              "isBackward": false,
              "hasFocus": false,
            },
          },
          "decorator": null,
          "directionMap": Immutable.OrderedMap {
            "cvvlm": "LTR",
          },
          "forceSelection": false,
          "inCompositionMode": false,
          "inlineStyleOverride": null,
          "lastChangeType": null,
          "nativelyRenderedContent": null,
          "redoStack": Immutable.Stack [],
          "selection": Immutable.Record {
            "anchorKey": "cvvlm",
            "anchorOffset": 0,
            "focusKey": "cvvlm",
            "focusOffset": 0,
            "isBackward": false,
            "hasFocus": false,
          },
          "treeMap": Immutable.OrderedMap {
            "cvvlm": Immutable.List [
              Immutable.Record {
                "start": 0,
                "end": 0,
                "decoratorKey": null,
                "leaves": Immutable.List [
                  Immutable.Record {
                    "start": 0,
                    "end": 0,
                  },
                ],
              },
            ],
          },
          "undoStack": Immutable.Stack [],
        },
      },
      "entity": Object {
        "getData": [Function],
        "getMutability": [Function],
        "getType": [Function],
      },
      "entityKey": "string",
      "entityType": Object {
        "description": "test",
        "icon": "test",
      },
      "lockEditor": [Function],
      "onChange": [Function],
      "onEditEntity": [Function],
      "onRemoveEntity": [Function],
      "textDirectionality": "RTL",
      "unlockEditor": [Function],
    }
  }
  isLoading={false}
  label="Test title"
  src="http://www.example.com/example.png"
>
  <a
    className="EmbedBlock__link"
    href="http://www.example.com/"
    rel="noopener noreferrer"
    target="_blank"
    title="http://www.example.com/"
  >
    Test title
  </a>
  <button
    className="Tooltip__button"
    onClick={[Function]}
    type="button"
  >
    Edit
  </button>
  <button
    className="Tooltip__button"
    onClick={[Function]}
    type="button"
  >
    Remove
  </button>
</MediaBlock>
`;
